src/flow-spec/IBlock.ts

```diff
@@ -107,21 +101,62 @@ export function findBlockExitWith(uuid: string, block: IBlock): IBlockExit {
   return exit
 }
 
-export function findFirstTruthyEvaluatingBlockExitOn(block: IBlockWithTestExits, context: IContext): IBlockExitTestRequired | undefined {
+/**
+ * @param block
+ * @param context
+ * @deprecated Use firstTrueBlockExitOrThrow or firstTrueBlockExitOrNull
+ */
+export function findFirstTruthyEvaluatingBlockExitOn(block: IBlock, context: IContext): IBlockExit | undefined {
   const {exits} = block
   if (exits.length === 0) {
     throw new ValidationException(`Unable to find exits on block ${block.uuid}`)
   }
 
   const evalContext = createEvalContextFrom(context)
-  return find<IBlockExitTestRequired>(
-    exits,
-    ({test, default: isDefault = false}) => !isDefault && evaluateToBool(String(test), evalContext)
-  )
+  return find<IBlockExit>(exits, ({test, default: isDefault = false}) => !isDefault && evaluateToBool(String(test), evalContext))
+}
+
+export function firstTrueBlockExitOrNull(block: IBlock, context: IContext): IBlockExit | undefined {
+  try {
+    return firstTrueBlockExitOrThrow(block, context)
+  } catch (e) {
+    return undefined
+  }
+}
+
+export function firstTrueBlockExitOrThrow(block: IBlock, context: IContext): IBlockExit {
+  const blockExit = _firstBlockExit(context, block)
+  if (blockExit == null) {
+    throw new ValidationException(`All block exits evaluated to false. Block: ${block.uuid}`)
+  }
+  return blockExit
+}
+
+function _firstBlockExit(context: IContext, block: IBlock): IBlockExit | undefined {
+  try {
+    const evalContext = createEvalContextFrom(context)
+    return (
+      find<IBlockExit>(block.exits, blockExit => evaluateToBool(String(blockExit.test), evalContext)) ?? findDefaultBlockExitOnOrNull(block)
```

Comment:
Do we need `?? findDefaultBlockExitOnOrNull(block)`?
If yes, should `firstTrueBlockExitOrThrow` better be called `firstTrueOrDefaultBlockExitOrThrow`?

Reply:
It's necessary, as almost none of the blocks follow the spec, and a block may be marked as `default`, but have no test, and we need to handle that case. We also need to be able to adapt as soon as the block becomes in spec, without breaking anything.